python/mage/graph_coloring_module/population_factory.py

```diff
@@ -0,0 +1,80 @@
+import logging
+from typing import Dict, Any, Optional, List
+from mage.graph_coloring_module.graph import Graph
+from mage.graph_coloring_module.components.population import Population
+from mage.graph_coloring_module.components.individual import Individual
+from mage.graph_coloring_module.components.chain_chunk import ChainChunk
+from mage.graph_coloring_module.components.chain_population import ChainPopulation
+from mage.graph_coloring_module.utils.parameters_utils import param_value
+from mage.graph_coloring_module.utils.validation import validate
+from mage.graph_coloring_module.exceptions import PopulationCreationException
+
+
+logger = logging.getLogger("telco")
+
+
+@validate("population_size", "no_of_colors")
+def generate_individuals(
+    graph: Graph, parameters: Dict[str, Any] = None
+) -> List[Individual]:
+    """Creates a list of individuals in which some individuals are the result of the given algorithms.
+    If more algorithms are given than the population size, then the excess is ignored."""
+
+    population_size = param_value(graph, parameters, "population_size")
+    no_of_colors = param_value(graph, parameters, "no_of_colors")
+    algorithms = param_value(graph, parameters, "algorithms")
+
+    individuals = []
+    if algorithms is not None:
+        for algorithm in algorithms:
+            indv = algorithm.run(graph, parameters)
+            if indv is None:
+                logger.error("Population creation has not succeeded.")
+                raise PopulationCreationException(
+                    "Population creation has not succeeded."
+                )
+            if len(individuals) < population_size:
+                individuals.append(indv)
+
+    individuals.extend(
+        [
+            Individual(no_of_colors, graph)
+            for _ in range(population_size - len(individuals))
+        ]
+    )
+    return individuals
+
+
+@validate("population_size", "no_of_chunks")
+def create(graph: Graph, parameters: Dict[str, Any] = None) -> Optional[Population]:
```

Comment:
Sry for the delay, forgot to push the comment, written it 3 days ago...

Reply:
Thanks for the reply. This is simply for the purpose of extensibility, meaning some other population method doesn't necessarily require `population_size` and `no_of_chunks` parameters. Also, both parameters are validated when entering the function's scope.

Reply:
It seems that for some components it is not easy to define a common interface. For example, an interface for mutation, because different implementations of the mutation have different parameters that we don't know in advance.